python/test/t_Gibbs_std.py

```diff
@@ -1,14 +1,55 @@
 #! /usr/bin/env python
-
 import openturns as ot
 import openturns.testing as ott
-import math as m
+import math
 
 ot.TESTPREAMBLE()
 
 ot.RandomGenerator.SetSeed(0)
 
-# this analytical example is taken from "Bayesian Modeling Using WinBUGS" - Ioannis Ntzoufras
+# Test the Metropolis within Gibbs algorithm to simulate a correlated 2D Gaussian target :
+
+# Define the target law
+mean_target = [0.0, 0.0]
+std_target = [1.0, 1.0]
+rho_target = 0.9
+target_normal = ot.Normal( mean_target, std_target, ot.CorrelationMatrix([[1., rho_target],[rho_target, 1.]]) )
+
+# Initial state of the Gibbs sampler
+initialState = mean_target
+
+# Define RWMH sampler for first coordinate
+proposal_std = 0.1
+proposal = ot.Normal(0.0, proposal_std)
+rwmh_sampler = ot.RandomWalkMetropolisHastings(target_normal, initialState, proposal, [0])
+
+# Define exacte sampler for second coordinate
+randomVector = ot.RandomVector(ot.Normal())
+
+# Define link function, implementing the Gaussian conditioning formulae
+# here a Python link function is used. TODO: use a symbolic link function instead 
+def py_link_function(x):
+    cond_mean = mean_target[1] + rho_target * std_target[1] * (x[0] - mean_target[0]) / std_target[0]
+    cond_std = std_target[1] * math.sqrt(1. -  rho_target**2)
+    return [cond_mean, cond_std]
+
+plf = ot.PythonFunction(2, 2, py_link_function)
+rv_sampler = ot.RandomVectorMetropolisHastings(randomVector, initialState, [1], plf)
+
+# Create Gibbs Sampler
+gibbs = ot.Gibbs([rwmh_sampler, rv_sampler])
+
+# Generate posterior distribution sample
+n = 10000
+Xsample = gibbs.getSample(n+rwmh_sampler.getBurnIn())[rwmh_sampler.getBurnIn():]
+
+# Compare empirical to theoretical moments
+
+ott.assert_almost_equal(Xsample.computeMean(), mean_target, 0.0, 10.0 / math.sqrt(n))
+ott.assert_almost_equal(Xsample.computeStandardDeviation(), std_target, 0.0, 10.0 / math.sqrt(n))
+ott.assert_almost_equal(Xsample.computeLinearCorrelation()[0,1], rho_target, 0.0, 10.0 / math.sqrt(n))
+
+# # this analytical example is taken from "Bayesian Modeling Using WinBUGS" - Ioannis Ntzoufras
 # 1.5.3: Inference for the mean or normal data with known variance
```

Comment:
Which page?

Reply:
@MerlinKeller I don't have the book, can you answer @mbaudin47 ?

Reply:
Page 9!

Reply:
So I suggest to insert that interesting information within the comments.

Reply:
OK, done